### ISSUE TITLE
Validierungsfehler bei mehrfacher Angabe von "ReceivableSpecifiedTradeAccountingAccount" im Extended-Profile

### ISSUE THREAD

**Issue by Alpini1980** [ZUGFeRD/mustangproject#279]
Hallo,

ich bin mir hier nicht sicher, ob es ein Problem im Mustangproject ist (ich vermute nicht), oder ich die ZUGFeRD-Doku falsch verstehe (gut möglich), oder es ein Bug in der Schematron-Datei oder der ZUGFeRD-Dokumentation ist:

Beim Validieren unserer Testrechnung (mit Extended-Profile) erhalte ich zwei Meldungstypen, die beide die gleiche Ursache haben - die mehrfache Angabe von receivablespecifiedtradeaccountingaccount in den Rechnungspositionen: 

Einmalig kommt diese Meldung (die Zeile 58 entspricht dem 2. Vorkommen des ram:ReceivableSpecifiedTradeAccountingAccount bei der ersten Rechnungsposition):
**2022-07-20 14:52:10.743 [main] ERROR o.m.validator.ZUGFeRDValidator - Error 18: schema validation fails:org.xml.sax.SAXParseException; lineNumber: 58; columnNumber: 56; cvc-complex-type.2.4.d: Invalid content was found starting with element 'ram:ReceivableSpecifiedTradeAccountingAccount'. No child element is expected at this point.**

Sich wiederholend für jede Rechnungsposition diese Meldung:
**2022-07-20 14:52:33.344 [main] INFO  o.m.validator.XMLValidator - FailedAssert 
2022-07-20 14:52:33.358 [main] ERROR o.m.validator.ZUGFeRDValidator - Error 4: 
	Element 'ram:ReceivableSpecifiedTradeAccountingAccount' may occur at maximum 1 times. (From /xslt/ZF_220/FACTUR-X_EXTENDED.xslt)**

Aus der ZUGFeRD-Dukumentation "ZUGFeRD-2.1.1 - Spezifikation_TA.pdf" lese ich zum "receivablespecifiedtradeaccountingaccount" auf Seite 53 Folgendes heraus:

**Datentyp: ram:TradeAccountingAccountType:
Kardinalität: 0 .. unbounded
Verwendung im Extended Profil zugelassen, ohne eine abweichende Kardinalität:**

Zusammenfassend würde ich meinen, dass die mehrfache Angabe des "receivablespecifiedtradeaccountingaccount" laut Doku im Extended-Profil erlaubt ist und wundere mich über die Validierungsfehler. Wir verwenden die doppelte Angabe des receivablespecifiedtradeaccountingaccount zur Angabe einer Kostenstelle sowie eines internen Buchungskontos des Kunden.

Ich würde mich freuen hier andere Meinungen zu hören.

Viele Grüße, Stefan Wagner

**Comment by jstaerk** [comment]
Hallo,
mit einem kompletten XMl-Beispiel kann ich das gerne mal upstreamen, nur ein Hinweis: im Extended-Profil ist das wirklich 0..n aber im EN16931-Profil ist es wohl 0..1 vgl 
![image](https://user-images.githubusercontent.com/3964386/180142384-fba4c2f6-6bdd-4b6d-af4e-842059ebce7c.png)

Da sich die validierermeldung allerdings bereits explizit auf Extended bezieht ist das wahrscheinlich tatsächlich ein Fehler in den Schematrons.

**Comment by msccip** [comment]
Hallo,
in ZUGFeRD 2.1 war es noch analog der Dokumentation deklariert. In 2.2 fehlt die Angabe maxOccurs="unbounded":

![grafik](https://user-images.githubusercontent.com/69585478/180168253-2e671dd5-982f-4684-a877-0bde40271407.png)

![grafik](https://user-images.githubusercontent.com/69585478/180168475-e15e7b2e-b57b-43e9-bdb4-9de91d2fbd6d.png)

s. jeweils FACTUR-X_EXTENDED_urn_un_unece_uncefact_data_standard_ReusableAggregateBusinessInformationEntity_100.xsd

**Comment by Alpini1980** [comment]
Hallo Jochen,

hier gezippt eine komplette .xml Anlage (anonymisiert) mit mehreren ReceivableSpecifiedTradeAccountingAccount's:

[zugferd-invoice.zip](https://github.com/ZUGFeRD/mustangproject/files/9158270/zugferd-invoice.zip)

Werden die Schematron-Dateien eigentlich auch in einem Github-Projekt gepflegt, so dass man es dorthin upstreamen bzw. als issue melden kann? Ansonsten schon einmal besten Dank für's weitermelden. Aber selbst wenn der Fehler korrigiert wird, frage ich mich in welcher Form das erfolgen würde. Bei einer Korrektur in Version 2.2 müsste ja jeder Verwender die Schema-Dateien neu herunterladen? 

@msccip: interessante Entdeckung, dass es in der älteren Version schon einmal mit der Doku konform war. Danke für den Hinweis.

Viele Grüße, Stefan Wagner

**Comment by jstaerk** [comment]
Hallo,
es könnte in den nächsten Monaten wohl ein ZUGFeRD 2.2.1 Version geben und ich kann nur hoffen dass es dabei korrigiert wird.
Meines Wissens wird der öffentliche Governance-Prozess überarbeitet, und das Gitlab ist derzeit leider auch noch nicht geöffnet, das heißt ein Upstream wäre nur per mail an die info@ferd-net.de möglich. Da die Schematron-Dateien aber generiert werden müsste aber selbst eine korrigierte Schematrondatei erstmal nachbearbeitet werden um nachhaltig zu werden.

**Comment by msccip** [comment]
@Alpini1980 
Ich habe mir Ihre Beispieldatei auch mal angesehen. Sie geben ja dort für jede Rechnungsposition dieselben 2 ReceivableSpecifiedTradeAccountingAccounts an. Ich weiß ja nicht, ob das immer so ist, aber falls ja, können Sie die Angabe auch auf Rechnungsebene machen. Dort ist auch unter 2.2 die Deklaration korrekt (maxOccurs="unbounded).

![grafik](https://user-images.githubusercontent.com/69585478/180213716-bb1ea19c-194f-4fc0-956e-9e1ed212300f.png)

**Comment by msccip** [comment]
auch interessant:
https://www.zugferd-community.net/forum/viewtopic.php?f=4&t=149

hier ist das Extended-Profil korrekt aber zumindest in EN16931 fehlt auch hier seit 2.2 maxOccurs="unbounded"

**Comment by Alpini1980** [comment]
...leider nein, das ist nicht immer so, sondern nur in dem stark gekürzten Beispiel. Das stimmt, in dem Beispiel könnte das über eine einmalige Angabe beider Felder im Header gelöst werden.

Im Normalfall haben unsere Rechnungen aber schnell 5000 Positionen pro Monat, die sich dann auf 50-100 Kostenstellen und bis zu 25 kundeninterne Sachkonten verteilen können. In Papier bzw. PDF-Form sind das dann schnell viele hunderte Seiten, in welchen dann pro Sachkonto und Kostenstelle Zwischensummen enthalten sind.

**Comment by Alpini1980** [comment]
> hier ist das Extended-Profil korrekt aber zumindest in EN16931 fehlt auch hier seit 2.2 maxOccurs="unbounded"

Ich kann den Link mangels Registrierung in dem Forum zwar nicht öffnen, aber da ohne maxOccurs-Angabe ja "1" als default genutzt wird, stimmt es ja dann für das EN 16931 Profil, da dieses ja die abweichende Kardinalität 0..1 hat. Oder verstehe ich etwas falsch?

**Comment by msccip** [comment]
in diesem Fall geht es die Bankverbindungen:

 <xs:element name="SpecifiedTradeSettlementPaymentMeans" type="ram:TradeSettlementPaymentMeansType" minOccurs="0"/>

**Comment by Alpini1980** [comment]
Diese sind aber ausschließlich dem "ApplicableHeaderTradeSettlement" untergeordnet und können nicht pro Position angegeben werden. Angegeben werden in den "SpecifiedTradeSettlementPaymentMeans" die verfügbaren Konten des Zahlungsempfängers,

**Comment by msccip** [comment]
@Alpini1980 
Ja, aber im EN16931 kann man entgegen der Spezifikation nur eine Bankverbindung angeben. Es ist letztlich derselbe Fehler, wie bei Ihrem Problem: es fehlt im 2.2 der maxOccurs="unbounded":

2.1 richtig (0..n):
![grafik](https://user-images.githubusercontent.com/69585478/180232789-ca0f1362-9bcf-48ef-aee4-cc29acfde31e.png)

2.2 falsch (0..1):
![grafik](https://user-images.githubusercontent.com/69585478/180232954-e75532cc-a32f-4963-8ee3-2fb72469a1b7.png)

**Comment by Alpini1980** [comment]
Sorry, jetzt habe ich erst verstanden worum es geht (und konnte es vorher nicht, ohne die Forenbeiträge zu sehen). Es wurde der gleiche Fehler noch bei einem weiteren Element (SpecifiedTradeSettlementPaymentMeans) gemacht und nun kann in der 2.2 beim EN 16931-Profil nur noch eine Bankverbindung angegeben werden, obwohl vorher eine unbegrenzte Anzahl erlaubt war.

**Comment by jstaerk** [comment]
Ja. Und das ist ein Fehler in den offiziellen Prüfdateien die von ZUGFeRD veröffentlicht werden und nicht im Mustang Validierer der sie nur anwendet (deren Issue 123). Da das ein vergleichsweise dramatischer Fehler ist gehe ich fast davon aus, dass die Behebung im entsprechenden Meeting am Freitag in einer Woche für ZUGFeRD 2.2.1 eingeplant werden wird.

**Comment by hvbtup** [comment]
Wie ist hier eigentlich der Stand?
Gibt es nach wie vor keine Möglichkeit, mehrere Bankverbindungen für SEPA-Überweisungen (Typ 58) oder sonstigen Überweisungen (Typ 30) anzugeben?

**Comment by jstaerk** [comment]
korrekt, hat das jemand mal gemeldet?
